src/Equations/elastic/Model/datastructures.hpp

```diff
@@ -37,16 +37,71 @@
  * @section DESCRIPTION
  **/
 
-#ifndef MODEL_DATASTRUCTURES_H_
-#define MODEL_DATASTRUCTURES_H_
+#ifndef MODEL_ELASTIC_DATASTRUCTURES_H_
+#define MODEL_ELASTIC_DATASTRUCTURES_H_
 
 #include <Model/common_datastructures.hpp>
+#include <cmath>
+#include <generated_code/kernel.h>
+#include <generated_code/init.h>
 
 namespace seissol {
   namespace model {
-    struct Material : public ElasticMaterial {};
-    struct LocalData {};
-    struct NeighborData {};
+    struct ElasticMaterial : Material {
+      double lambda;
+      double mu;
+
+      virtual ~ElasticMaterial() {};
+
+      void getFullElasticTensor(real fullTensor[81]) const final {
+
+        auto elasticTensorView = init::mElasticTensor::view::create(fullTensor);
+        elasticTensorView.setZero();
+        elasticTensorView(0,0,0,0) = lambda + 2*mu;
+        elasticTensorView(0,0,1,1) = lambda;
+        elasticTensorView(0,0,2,2) = lambda;
+        elasticTensorView(0,1,0,1) = mu;
+        elasticTensorView(0,1,1,0) = mu;
+        elasticTensorView(0,2,0,2) = mu;
+        elasticTensorView(0,2,2,0) = mu;
+        elasticTensorView(1,0,0,1) = mu;
+        elasticTensorView(1,0,1,0) = mu;
+        elasticTensorView(1,1,0,0) = lambda;
+        elasticTensorView(1,1,1,1) = lambda + 2*mu;
+        elasticTensorView(1,1,2,2) = lambda;
+        elasticTensorView(1,2,1,2) = mu;
+        elasticTensorView(1,2,2,1) = mu;
+        elasticTensorView(2,0,0,2) = mu;
+        elasticTensorView(2,0,2,0) = mu;
+        elasticTensorView(2,1,2,1) = mu;
+        elasticTensorView(2,2,0,0) = lambda;
+        elasticTensorView(2,2,1,1) = lambda;
+        elasticTensorView(2,2,2,2) = lambda + 2*mu;
+      }
+
+      double getMaxWaveSpeed() const final {
+        return getPWaveSpeed();
+      }
+
+      double getPWaveSpeed() const final {
+        return std::sqrt((lambda + 2*mu) / rho);
+      }
+
+      double getSWaveSpeed() const final {
+        return std::sqrt(mu / rho);
+      }
+
+      MaterialType getMaterialType() const override{
+        return MaterialType::elastic;
+      }
+    };
+
+    struct ElastoPlasticMaterial : ElasticMaterial, Plasticity {
+      virtual ~ElastoPlasticMaterial() {};
+      MaterialType getMaterialType() const final {
+        return MaterialType::elastoplastic;
+      }
+    };
```

Comment:
I really don't like this multiple inheritance.
It doesn't make sense anyway, why is ElastoPlasticMaterial a subtype of Plasticity?